alttpo/GameState.as

```diff
@@ -1112,9 +1137,256 @@ class GameState {
+        }
+        tile.draw_sprite_4bpp(24, 32, p, sprs[27], palette);
+        tile.draw_sprite_4bpp(16, 32, p, sprs[11], palette);
+        tile.draw_sprite_4bpp(8, 32, p, sprs[26], palette);
+    }
+    // morph ball
+    else if (protocol == 4){
+        for (int i = 0; i < 4; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4), p, sprs[i], palette);
+        }
+        for (int i = 0; i < 4; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4)+8, p, sprs[i+16], palette);
+        }
+        tile.draw_sprite_4bpp(16, 16, p, sprs[5], palette);
+        tile.draw_sprite_4bpp(8, 16, p, sprs[20], palette);
+    }
+    //spin jump
+    else if (protocol == 5){
+        for (int i = 0; i < 16; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4), p, sprs[i], palette);
+        }
+        for (int i = 0; i < 16; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4)+8, p, sprs[i+16], palette);
+        }
+    }
+    //facing toward the camera
+    else if (protocol == 6){
+        for (int i = 0; i < 12; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4), p, sprs[i], palette);
+        }
+        for (int i = 0; i < 12; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4)+8, p, sprs[i+16], palette);
+        }
+        tile.draw_sprite_4bpp(0, 48, p, sprs[12], palette);
+        tile.draw_sprite_4bpp(16, 48, p, sprs[13], palette);
+        tile.draw_sprite_4bpp(8, 48, p, sprs[28], palette);
+        tile.draw_sprite_4bpp(24, 48, p, sprs[29], palette);
+    }
+    // vertical leap
+    else if (protocol == 7){
+        for (int i = 0; i < 12; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4), p, sprs[i], palette);
+        }
+        for (int i = 0; i < 12; i++){
+            tile.draw_sprite_4bpp(8*(i%4), 16*(i/4)+8, p, sprs[i+16], palette);
+        }
+        tile.draw_sprite_4bpp(16, 48, p, sprs[13], palette);
+        tile.draw_sprite_4bpp(8, 48, p, sprs[28], palette);
+    }
+    
+    return ei;
+  }
+  
+  int determine_protocol(uint8 pose){
+    switch(pose){
+        case 0x00: return 6;
+        case 0x03: return 1;
+        case 0x04: return 1;
+        case 0x27: return 2;
+        case 0x28: return 2;
+        case 0x38: return 3;
+        case 0x3e: return 3;
+        case 0x43: return 3;
+        case 0x44: return 3;
+        case 0x4b: return 7;
+        case 0x4c: return 7;
+        case 0x4d: return 7;
+        case 0x4e: return 7;
+        case 0x71: return 3;
+        case 0x72: return 3;
+        case 0x73: return 3;
+        case 0x74: return 3;
+        case 0x79: return 4;
+        case 0x7a: return 4;
+        case 0x7b: return 4;
+        case 0x7c: return 4;
+        case 0x7d: return 4;
+        case 0x7e: return 4;
+        case 0x7f: return 4;
+        case 0x80: return 4;
+        case 0x81: return 5;
+        case 0x82: return 5;
+        case 0xa4: return 7;
+        case 0xa5: return 7;
+        default: return 0;
```

Comment:
Are you missing 0x19 and 0x1a poses for spin jump here?

Reply:
for whatever reason those poses appear fine with the default protocol, so i didn't need to list them

Reply:
oh really? i'm guessing it depends on what equipment you have. i know i saw 19 and 1a poses on a clean startup without screw attack